packages/core/src/index.ts

```diff
@@ -1196,6 +1196,10 @@ class Hopr extends EventEmitter {
     return await this.connector.getPublicKeyOf(addr)
   }
 
+  public async getEntryNodes(): Promise<{ id: PeerId; multiaddrs: Multiaddr[] }[]> {
+    return this.connector.waitForPublicNodes()
```

Comment:
You don't need run that async call here, instead you can use `this.knownPublicNodesCache`

Reply:
Don't expect this command to run very often since it's meant for debugging, so I wouldn't do many optimizations. I anyway expect that checking the eligibility is the main bottleneck of it.

Apart from that, the db holds the ultimate truth, which can affect a lot of functionalities, and which is why I prefer to fetch the entry nodes from the db.